sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java

```diff
@@ -146,6 +147,16 @@ public TableServiceAsyncClient buildAsyncClient() {
 
         AzureNamedKeyCredential namedKeyCredential = null;
 
+        // If 'endpoint' was provided and reflects Storage acct., validate its format to end with the appropriate suffix
+        if (endpoint != null && endpoint.contains(suffix)) {
```

Comment:
If we are using other cloud environments - like US gov, China, Azure Stack etc. the suffix can be different. So, this check may not work for non-public cloud environments .